### PR TITLE
added test cases and made code improvements

### PR DESCRIPTION
**Pull request by owaisrazaeng1**

### CODE REVIEW

File: spec/system/device_spec.rb [hunk 1]
```diff
@@ -0,0 +1,31 @@
+require 'rails_helper'
```
Comment (olivierobert) [inline spec/system/device_spec.rb:1]: Did you mean to name the file "devi**s**e_spec"

File: spec/system/device_spec.rb [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+    fill_in "user_first_name", with: "Harry"
+    fill_in "user_last_name", with: "Potter"
+    fill_in "user_email", with: "hp2@gmail.com"
+    fill_in "user_password", with: "Admin@123"
+    fill_in "user_password_confirmation", with: "Admin@123"
+
+
+    click_button 'Sign up'
+    expect(page).to have_content 'Sign Out'
```
Comment (olivierobert) [inline spec/system/device_spec.rb:19]: How about checking the page is redirected to the dashboard instead?

File: app/models/search_query.rb [hunk 3]
```diff
@@ -8,8 +8,14 @@ class SearchQuery < ApplicationRecord
...
-      keywords << row[0]
+    CSV.foreach(file.path) do |row|
+      if row.count > 1
+        row.each do |keyword|
+          keywords << keyword
+        end
+      else
+        keywords << row[0]
+      end
     end
```
Comment (olivierobert) [inline app/models/search_query.rb:19]: There needs to also be a validation for not allowing more than 100 keywords as per the requirements

File: spec/system/device_spec.rb (hunk 1 above)
Comment (owaisrazaeng1) [reply spec/system/device_spec.rb:1]: resolved in https://github.com/owaisrazaeng1/nimble-test/commit/7aa833bd51593f376ce1c36a37c31e7b314e627f

File: spec/system/device_spec.rb (hunk 2 above)
Comment (owaisrazaeng1) [reply spec/system/device_spec.rb:19]: resolved in https://github.com/owaisrazaeng1/nimble-test/commit/7aa833bd51593f376ce1c36a37c31e7b314e627f

File: app/models/search_query.rb (hunk 3 above)
Comment (owaisrazaeng1) [reply app/models/search_query.rb:19]: resolved in https://github.com/owaisrazaeng1/nimble-test/commit/7aa833bd51593f376ce1c36a37c31e7b314e627f

**Pull request merged**